src/vm/virtualMachine.ts

```diff
@@ -125,381 +57,292 @@ export class VirtualMachine {
 
     const encodedInstruction = maybeEncodedInstruction.toBigInt();
 
-    // decode and run instruction
     const instruction = Instruction.decodeInstruction(encodedInstruction);
 
-    return this.runInstruction(instruction);
+    this.runInstruction(instruction);
```

Comment:
so `step` is basically `decode + run` and ` decode` belongs to `Instruction`. I'd then inline the `run` here

Reply:
`step` will later execute hints before handling the instruction, we should keep it as is